app/utils/universalRaven.js

```diff
@@ -0,0 +1,28 @@
+//@flow
+
+export interface UniversalRaven {
```

Comment:
couldn't this be replaced with [raven-node](https://docs.sentry.io/clients/node/) if we only use it on the server anyway?

Reply:
**TL;DR**: Nope...

Because we use `react-prepare` and a lot of promises to do things, it is impossible to handle it directly using raven node. The promises "forks" away from the main thread, making it impossible for raven-node to understand what is connected to what. The reason we can use `raven` like we does in the client, is because is uses the global scope to store breadcrumbs/contexts etc. This approach works by binding the `UniversalRavenNode` functions to the context defined here (using the implicit binding of arrow functions): https://github.com/webkom/lego-webapp/pull/1034/files#diff-bec864430eb5752a683c8798d0c6bd6cR79. Then it passes the "custom" raven instance to all the components that use it during ssr.

I have used way too much time on this PR now :laughing: Still having some issues with sourcemaps when the `ENVIRONMENT` isn't development, so I guess I have to investigate it some more..


__PS__: The `RavenOptions` is from the flow typeing you mentioned over. :smile: